content/articles/creating-serverless-functions-faas-with-aws-lambda-and-golang/index.md

```diff
@@ -0,0 +1,292 @@
+
+### Calling the Lambda Function From Golang
+
+We can also execute the function uploaded to AWS lambda locally. We have to install the AWS Go SDK.
+
+```bash
+
+go get github.com/aws/aws-sdk-go
+
+```
+
+We can now use this code segment to execute the function in the cloud and see the results locally.
+
+```go
+
+package main
+
+ 
+
+import (
+
+ "encoding/json"
+
+ "fmt"
+
+ "os"
+
+ 
+
+ "github.com/aws/aws-sdk-go/aws"
+
+ "github.com/aws/aws-sdk-go/service/lambda"
+
+ "github.com/joho/godotenv"
+
+ 
+
+ "github.com/aws/aws-sdk-go/aws/credentials"
+
+ "github.com/aws/aws-sdk-go/aws/session"
+
+ // "github.com/aws/aws-lambda-go/lambda"
+
+)
+
+ 
+
+type Info struct {
+
+ Firstname string `json:"firstname"`
+
+ Lastname string `json:"lastname"`
+
+ Age int `json:"age"`
+
+}
+
+ 
+
+type Response struct {
+
+ Profile string `json:"profile"`
+
+}
+
+ 
+
+func main() {
+
+ // lambda.Start(HandleLambdaEvent)
+
+ // event := MyEvent{Name: "Tomi", Age: 17}
+
+ // c, err := json.Marshal(event)
+
+ // if err != nil {
+
+ // fmt.Println("Error")
+
+ // }
+
+ godotenv.Load()
+
+ sess := session.Must(session.NewSessionWithOptions(session.Options{SharedConfigState: session.SharedConfigEnable}))
+
+ client := lambda.New(sess, &aws.Config{Region: aws.String("us-east-1"), Credentials: credentials.NewStaticCredentials(os.Getenv("aws_access_key_id"), os.Getenv("aws_secret_access_key"), "")})
+
+ request := Info{Firstname: "Oluwatomisin", Lastname: "Bamimore", Age: 16}
+
+ payload, err := json.Marshal(request)
+
+ if err != nil {
+
+ fmt.Println("Error marshalling request")
+
+ os.Exit(0)
+
+ }
+
+ result, err := client.Invoke(&lambda.InvokeInput{FunctionName: aws.String("user-profile"), Payload: payload})
+
+ if err != nil {
+
+ fmt.Println("Error calling user-profile function", err)
+
+ os.Exit(0)
+
+ }
+
+ var resp Response
+
+ err = json.Unmarshal(result.Payload, &resp)
+
+ if err != nil {
+
+ fmt.Println("Error unmarshalling user-profile response")
+
+ os.Exit(0)
+
+ }
+
+ 
+
+ fmt.Println(resp.Profile)
+
+}
+
+```
+
+You need to create an [IAM](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_users_create.html) user with the correct permissions. Store the `aws_access_key_id` and the `aws_secret_access_key` in a `.env` file in the directory you are currently working in.
+
+![sample-env-file](/engineering-education/creating-serverless-functions-faas-with-aws-lambda-and-golang/sample-env-file.png)After running the program, AWS will execute the cloud function, and you will get this result.
+
+![sample-program-output](/engineering-education/creating-serverless-functions-faas-with-aws-lambda-and-golang/sample-program-output.png)
+
+The program passes in some arguments into the cloud function and returns the result.
+
+### Conclusion
+
+We have successfully deployed and tested our AWS Lambda function with Golang. The same method may be used to build serverless functions in other languages such as Python, JavaScript, and Java.
```

Comment:
This is a very short conclusion.

A good conclusion summarizes the whole article, states the key takeaways from the article, has necessary reference links that the reader can refer.

Reply:
I have added something more to the conclusion. Thank you.